testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveViews.java

```diff
@@ -172,4 +172,88 @@ public void testLateralViewJsonTupleAs()
+                "  CAST(2592000.123 AS decimal(10,3)) source_decimal_three," +
+                "  CAST(2592000 AS DECIMAL(10,0)) source_decimal_zero," +
+                "  timestamp '1970-01-30 16:00:00' source_timestamp, " +
+                "  date '1970-01-30 16:00:00' source_date ");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS test_from_utc_timestamp_view");
+        onHive().executeQuery("CREATE VIEW " +
+                "test_from_utc_timestamp_view " +
+                "AS SELECT " +
+                // TODO: TZ identifiers like 'PST' is not supported by Trino; possibly we need a translation map (???)
+                "   from_utc_timestamp(source_tinyint, 'America/Los_Angeles') ts_tinyint, " +
+                "   from_utc_timestamp(source_smallint, 'America/Los_Angeles') ts_smallint, " +
+                "   from_utc_timestamp(source_integer, 'America/Los_Angeles') ts_integer, " +
+                "   from_utc_timestamp(source_bigint, 'America/Los_Angeles') ts_bigint, " +
+                "   from_utc_timestamp(source_float, 'America/Los_Angeles') ts_float, " +
+                "   from_utc_timestamp(source_double, 'America/Los_Angeles') ts_double, " +
+                "   from_utc_timestamp(source_decimal_three, 'America/Los_Angeles') ts_decimal_three, " +
+                "   from_utc_timestamp(source_decimal_zero, 'America/Los_Angeles') ts_decimal_zero, " +
+                "   from_utc_timestamp(source_timestamp, 'America/Los_Angeles') ts_timestamp, " +
+                "   from_utc_timestamp(source_date, 'America/Los_Angeles') ts_date " +
+                "FROM test_from_utc_timestamp_source");
+
+        // check result on Trino
+        assertThat(query("SELECT " +
+                "  CAST(ts_tinyint as VARCHAR), " +
+                "  CAST(ts_smallint as VARCHAR), " +
+                "  CAST(ts_integer as VARCHAR), " +
+                "  CAST(ts_bigint as VARCHAR), " +
+                "  CAST(ts_float as VARCHAR), " +
+                "  CAST(ts_double as VARCHAR), " +
+                "  CAST(ts_decimal_three as VARCHAR), " +
+                "  CAST(ts_decimal_zero as VARCHAR), " +
+                "  CAST(ts_timestamp as VARCHAR), " +
+                "  CAST(ts_date as VARCHAR) " +
+                "FROM test_from_utc_timestamp_view"))
+                .containsOnly(row(
+                        "1969-12-31 16:00:00.123",
+                        "1969-12-31 16:00:10.123",
+                        "1969-12-11 22:57:12.827",
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.000",
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.123",
+                        "1970-01-30 16:00:00.000",
+                        "1970-01-30 08:00:00.000",
+                        "1970-01-29 16:00:00.000"));
+
+        // check result on Hive
+        assertThat(onHive().executeQuery("SELECT " +
```

Comment:
There's a useful `assertViewQuery` which may reduce some repetition here (by moving the CAST to STRING into the view definition itself).

Reply:
I tried to use that but it does not help as the string values generated by Trino and Hive are different for timestamps (Hive truncates zeros).